### ISSUE TITLE
Feature Request: Read Only User Account creation option

### ISSUE THREAD

**Issue by coagen** [partkeepr/PartKeepr#385]
This would be really handy for my use-case.

Thanks!

**Comment by kitusmark** [comment]
It seems that is possible to set some permissions in the user page, but I've no idea how to do it. 
If you go to the demo site in www.demo.partkeepr.org you'll see the attributes.

Something like this:
![partkeepr](https://cloud.githubusercontent.com/assets/6075056/7535899/6442d89c-f58a-11e4-86b0-4ee3ef6d74fa.PNG)

I ***really*** need this feature. Anyone knows something??

Thanks

**Comment by felicitus** [comment]
The only thing you can set is administrative rights, which simply enables/disables user management but no further permission management.

The attributes you see are indirectly set by the preferences, but they are not permissions of any kind.

If you need read-only access, you might try to create a separate PartKeepr instance which points to the same database, but with a database read-only user. However, I didn't try this.

For PartKeepr 1.0, I'm looking forward implementing Symfony [ACLs](http://symfony.com/doc/current/cookbook/security/acl.html) but this will take quite some time to be finalized.

**Comment by DIYsciBorg** [comment]
Just seconding the desire for more granular access permissions. In particular:
1: view only account permission.  For people who need to locate parts, verify stock inventories, check footprint type, view datasheets, or compare pricing.

2: View and adjust inventories only account permission so that users can remove parts from inventory, verify and adjust counts, or log received stockups

3: Stock order permission, where the user can not adjust, but can generate order lists by distributor of all stock items below the minimum stock level and select quantities to order based on the best rates. (I understand the action here is actually a separate issue, but he permission would also have to exist).

**Comment by kodaman2** [comment]
Could this feature be implemented with voters? [https://symfony.com/doc/2.0/components/security/authorization.html#voters](url)

**Comment by Drachenkaetzchen** [comment]
Yes, but it's not as simple as including the library. Still very big effort to first upgrade the whole PartKeepr infrastructure to the latest symfony version (or 3.x)

**Comment by kodaman2** [comment]
I could help as needed, not that I know much about symfony but would like to help in the migration process. I'll read the docs on symfony. I can branch out and start on some bundles, and go about it that way.

**Comment by Drachenkaetzchen** [comment]
The migration branch is at https://github.com/partkeepr/PartKeepr/tree/test-dunglas-migration
